service-csw/src/main/java/fi/nls/oskari/csw/helper/CSWISORecordDataQualityParser.java

```diff
@@ -90,64 +91,74 @@ public CSWISORecordDataQualityParser() {
             XPATH_QUANTITATIVE_RESULT_VALUE_UNIT = xpath.compile("./gmd:valueUnit");
             XPATH_QUANTITATIVE_RESULT_ERROR_STATISTIC = xpath.compile("./gmd:errorStatistic"); //many
             XPATH_QUANTITATIVE_RESULT_VALUE = xpath.compile("./gmd:value");
+
+            //Free text (character string)
+            XPATH_CHARACTER_STRING = xpath.compile("./gco:CharacterString");
         }
         catch (Exception e) {
             log.error("Setting XPaths failed in data quality parser");
             throw new RuntimeException("Setting XPaths failed in data quality parser");
         }
     }
 
-    public CSWIsoRecord.DataQualityObject parseDataQualities(final NodeList dataQualityNodes, final Locale locale)  throws XPathExpressionException {
-        if (locale != null) {
-                pathToLocalizedValue = xpath.compile(
-                        "../gmd:PT_FreeText/gmd:textGroup/gmd:LocalisedCharacterString[@locale='#"
-                                + locale.getLanguage().toUpperCase() + "']");
-        }
+    public CSWIsoRecord.DataQualityObject parseDataQualities(final NodeList dataQualityNodes, final XPathExpression pathToLoc)  throws XPathExpressionException {
+        pathToLocalizedValue = pathToLoc;
+
         CSWIsoRecord.DataQualityObject dataQualityObject = new CSWIsoRecord.DataQualityObject();
-        List<CSWIsoRecord.DataQualityNode> dataQualityObjectNodeList = dataQualityObject.getDataQualityNodes();
+        List<CSWIsoRecord.DataQuality> dataQualityList = dataQualityObject.getDataQualities();
```

Comment:
For readability I'd just created a `new ArrayList<T>` instead of calling the `#getDataQualities()`

Reply:
This is not addressed even though it shows up as outdated.